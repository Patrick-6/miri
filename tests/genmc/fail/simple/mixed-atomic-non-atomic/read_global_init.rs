//@compile-flags: -Zmiri-genmc -Zmiri-disable-stacked-borrows

#![no_main]

use std::sync::atomic::{AtomicU64, Ordering};

static X: AtomicU64 = AtomicU64::new(1234);

#[unsafe(no_mangle)]
fn miri_start(_argc: isize, _argv: *const *const u8) -> isize {
    // TODO GENMC: make this a "pass" test
    if 1234 != unsafe { *X.as_ptr() } {
        unsafe { std::hint::unreachable_unchecked() };
    }
    if 1234 == X.load(Ordering::SeqCst) {
        unsafe { std::hint::unreachable_unchecked() }; //~ ERROR: entering unreachable code
    }

    0
}
