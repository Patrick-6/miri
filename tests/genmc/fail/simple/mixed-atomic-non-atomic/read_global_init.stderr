Running GenMC Verification...
error: Undefined Behavior: entering unreachable code
  --> tests/genmc/fail/simple/mixed-atomic-non-atomic/read_global_init.rs:LL:CC
   |
LL |         unsafe { std::hint::unreachable_unchecked() };
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Undefined Behavior occurred here
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: BACKTRACE:
   = note: inside `miri_start` at tests/genmc/fail/simple/mixed-atomic-non-atomic/read_global_init.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

note: add `-Zmiri-genmc-print-genmc-output` to MIRIFLAGS to see the detailed GenMC error report

error: aborting due to 1 previous error

